src/input/gamepad/gamepadInfo.js

```diff
@@ -0,0 +1,58 @@
+// @flow
+/*eslint indent:0*/
+
+import {Vec2D} from "../../main/util/Vec2D";
+
+export type StickCardinals = { center : Vec2D, left : number, right : number, up : number, down : number };
+
+export type ButtonInfo = null 
+                       | { kind : "pressed", index : number }
+                       | { kind : "value", index : number, threshold : number }
+                       | { kind : "axis", index : number, threshold : number }
+export type StickInfo = null 
+                      | { kind : "axes", xIndex : number, yIndex : number, cardinals : null | StickCardinals }
+                      | { kind : "value", xIndex : number, yIndex : number, cardinals : null | StickCardinals }
+export type TriggerInfo = null 
+                        | { kind : "axis", index : number, min : number, max : number }
+                        | { kind : "value", index : number, min : number, max : number }
+                        | { kind : "light", index : number }
+export type DPadInfo =  null
+                     | { kind : "buttons", upIndex : number, downIndex : number, leftIndex : number, rightIndex : number }
+                     | { kind : "axis", index : number }
+                     | { kind : "2axes", xIndex : number, yIndex : number, xFlip : bool, yFlip : bool }
+
+export type GamepadID = { name : string, id? : string, vendor? : string, product? : string }
```

Comment:
Yeah, apparently in Linux (for some users at least) the string and product values are undefined.